### ISSUE TITLE
Link prefetching not working

### ISSUE THREAD

**Issue by maxsalven** [vercel/next.js#53518]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.12.1
      npm: 9.7.1
      Yarn: 1.22.19
      pnpm: N/A
    Relevant Packages:
      next: 13.4.13-canary.12
      eslint-config-next: 13.4.12
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router

### Link to the code that reproduces this issue or a replay of the bug

https://store-next-five.vercel.app/apple/ipod

### To Reproduce

Visit https://store-next-five.vercel.app/apple/ipod

In Chrome's developer tools network settings, you can see the `iphone` route gets prefetched.

Now set your Network Speed to 'Slow 3G', and click 'iPhone'. The prefetched data isn't used, you have to wait a full 2s before the new page renders.

The only code is at `app/[store]/[product]/page.tsx`:

```javascript
import Link from "next/link";

export default function Product({
  params,
}: {
  params: { store: string; product: string };
}) {
  return (
    <div>
      Product Page {params.store}/{params.product}
      <div>
        <Link href={`/${params.store}/ipod`}>iPod</Link>
      </div>
      <div>
        <Link href={`/${params.store}/iphone`}>iPhone</Link>
      </div>
    </div>
  );
}
```

### Describe the Bug

The prefetched data for the Link isn't used, and we have to wait for a full client-server request before we can render the new page.

### Expected Behavior

I expect clicking on the link to render immediately, even on a slow network connection, if the prefetched data has loaded.

**Comment by maxsalven** [comment]
Duplicate of https://github.com/vercel/next.js/issues/47981
